<!DOCTYPE html>
<html lang="en" dir="ltr">

<head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <title>The missing link: discerning true from false negatives when sampling species interaction networks</title>
    <script src="https://hypothes.is/embed.js" async></script>
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/normalize/8.0.1/normalize.min.css"
    integrity="sha256-l85OmPOjvil/SOvVt3HnSSjzF1TUMyT9eV0c2BzEGzU=" crossorigin="anonymous" />
    <style>
        @import url('https://fonts.googleapis.com/css2?family=Roboto:ital,wght@0,300;0,400;0,500;1,300;1,400;1,500&display=swap');
        @import url('https://fonts.googleapis.com/css2?family=Roboto+Condensed:ital,wght@0,300;0,400;0,700;1,300;1,400;1,700&display=swap');
        @import url('https://fonts.googleapis.com/css2?family=Roboto+Mono:ital,wght@0,400;0,500;1,400;1,500&display=swap');

        :root {
            --citation: #1976D2;
            --structure: hsl(210, 61%, 43%);
            
            --light: #474747;
            --pale: #a5a5a5;
            --grey: #e0e0e0;
            --nav: #fdfdfd;

            --width: 820px;
            --inner-space: 1.2em;
        }

        body {
            font-family: "Roboto", sans-serif;
            margin: 0 0;
            padding: 0 0;
        }

        article > p {
            line-height: 135%;
        }
        article > h1 {
            padding-top: 3em;
        }
        a {
            text-decoration: none;
            color: #000;
            border-bottom: 1px solid var(--structure);
        }
        a:hover {
            font-weight: 500;
        }
        .citation {
            color: var(--citation);
        }
        .container {
            margin: 0em auto;
            max-width: var(--width);
            padding: var(--inner-space);
        }
        header > .container {
            margin-top: 4em;
        }
        code, pre {
            font-family: "Roboto Mono", monospace;
        }
        pre {
            white-space: pre-wrap;
        }
        h1, h2, h3 {
            color: var(--structure);
            font-family: "Roboto Condensed", sans-serif;
            font-weight: 400;
        }
        header {
            font-family: "Roboto Condensed", sans-serif;
            z-index: 5;
            color: black;
            margin-top: 4rem;
        }
        .header--title {
            font-weight: 400;
            font-size: 2rem;
            margin-bottom: 1rem;
        }
        .header--authors > span {
            color: var(--light);
            font-weight: 400;
        }
        main {
            padding-top: 8em !important;
            padding-bottom: 8em !important;
        }
        footer {
            background-color: var(--grey);
            font-weight: 300;
        }
        .abstract {
            color: var(--light);
            margin-bottom: 5em;
        }
        figure {
            max-width: 100%;
            border: 1px solid var(--pale);
            border-radius: 2px;
            margin: 1em 0.5em;
        }
        figure > img {
            max-width: 100%;
            display: block;
            margin: 0em auto;
        }
        figure > figcaption {
            padding: var(--inner-space);
        }
        table {
            width: 100%;
            min-width: 100%;
            border-collapse: collapse;
        }
        table > caption {
            padding: var(--inner-space);
        }
        figcaption, caption {
            color: var(--light);
            font-weight: 300;
        }
        thead {
            color: var(--structure);
        }
        td, th {
            padding: 3px;
        }
        tr {
            border-top: 1px solid var(--structure);
            border-bottom: 1px solid var(--structure);
        }
        tbody > tr:nth-child(even) {background: var(--grey)};
        tbody {
            border-bottom: 1px solid var(--structure);
        }
        #refs {
            font-size: 90%;
            column-count: 2;
        }
        #refs > .csl-entry {
            padding-bottom: 0.2em;
            padding-top: 0.2em;
        }
        .hanging-indent > .csl-entry {
            padding-left: 15px ;
            text-indent: -15px ;
        }    </style>
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.10.0-rc.1/dist/katex.min.css"
    integrity="sha384-D+9gmBxUQogRLqvARvNLmA9hS2x//eK1FhVb9PiU86gmcrBrJAQT8okdJ4LMp2uv" crossorigin="anonymous" />
    <script defer src="https://cdn.jsdelivr.net/npm/katex@0.10.0-rc.1/dist/katex.min.js"
    integrity="sha384-483A6DwYfKeDa0Q52fJmxFXkcPCFfnXMoXblOkJ4JcA8zATN6Tm78UNL72AKk+0O"
    crossorigin="anonymous"></script>
    <script defer src="https://cdn.jsdelivr.net/npm/katex@0.10.0-rc.1/dist/contrib/auto-render.min.js"
    integrity="sha384-yACMu8JWxKzSp/C1YV86pzGiQ/l1YUfE8oPuahJQxzehAjEt2GiQuy/BIvl9KyeF"
    crossorigin="anonymous"></script>
    <script>
        document.addEventListener("DOMContentLoaded", function () {
            renderMathInElement(document.body, {
                delimiters: [
                { left: "$$", right: "$$", display: true },
                { left: "$", right: "$", display: false }
                ]
            });
        });
    </script>
</head>

<body>
    <header>
        <div class="container">
            <div class="header--title">The missing link: discerning true from false negatives when sampling species interaction networks</div>
            <div class="header--authors">
                <span class="header--author">
                <a href="https://orcid.org/0000-0002-6506-6487" target="_blank">
                    Michael D. Catchen
                </a>
                <sup class="affilcode">1</sup>
                <sup class="affilcode">2</sup>
                </span><span class="header--author">
                <a href="https://orcid.org/0000-0002-0735-5184" target="_blank">
                    Timothée Poisot
                </a>
                <sup class="affilcode">3</sup>
                <sup class="affilcode">2</sup>
                </span><span class="header--author">
                <a href="https://orcid.org/0000-0002-6004-4027" target="_blank">
                    Laura J. Pollock
                </a>
                <sup class="affilcode">1</sup>
                <sup class="affilcode">2</sup>
                </span><span class="header--author">
                <a href="https://orcid.org/0000-0001-6075-8081" target="_blank">
                    Andrew Gonzalez
                </a>
                <sup class="affilcode">1</sup>
                <sup class="affilcode">2</sup>
                </span>
            </div>
        </div>
    </header>
    
    <main class="container">
        <section class="abstract">
            


                    {\bfseries Abstract:}\,Ecosystems are composed of networks of interacting species. These interactions allow communities of species to persist through time through both neutral and adaptive processes. Despite their importance, a robust understanding of (and ability to predict and forecast) interactions among species remains elusive. This knowledge-gap is largely driven by a shortfall of data—although species occurrence data has rapidly increased in the last decade, species interaction data has not kept pace, largely due to the effort required to sample interactions. This means there are many interactions between species that occur in nature, but we do not know these interactions occur because we have never observed them. These so-called “false-negatives” bias data and hinder inference about the structure and dynamics of interaction networks. Here, we show the realized number of false-negatives in data can be quite high, even in thoroughly sampled systems, due to variation in abundances in a community. We provide a null model of occurrence detection to estimate the false-negative rate in a given dataset. We also show how to directly incorporate uncertainty due to observation error into model-based predictions of interactions between species. One hypothesis is interactions between “rare” species are themselves rare because these species are less likely to encounter one-another than species of higher relative abundance, and this can (in part) explain the common pattern of nestedness in bipartite interaction networks. However, we demonstrate that across several datasets of spatial/temporally replicated networks, there are positive associations between species co-occurrence and interactions, which suggests these interactions among “rare” species actually exist but simply are not observed. Finally, we assess how false negatives influence various models of network prediction, and recommend directly accounting for observation error in predictive models. We conclude by discussing how the understanding of false-negatives can inform how we design monitoring schemes for species interaction surveys.\\%
                
            <div class="artifacts">
                <a href="draft.pdf" title="Draft PDF" target="_blank">draft</a>
                <a href="preprint.pdf" title="Preprint PDF" target="_blank">preprint</a>
            </div>
        </section>
        
        <article>
            <h1 id="introduction">Introduction</h1>
            <p>Species interactions drive many processes in evolution and ecology. A better understanding of species interactions is an imperative to understand the evolution of life on Earth, to mitigate the impacts of anthropogenic change on biodiversity <span class="citation" data-cites="Makiola2020KeyQue">(Makiola <em>et al.</em> 2020)</span>, and for predicting zoonotic spillover of disease to prevent future pandemics <span class="citation" data-cites="Becker2021OptPre">(Becker <em>et al.</em> 2021)</span>. At the moment we lack sufficient data to meet these challenges <span class="citation" data-cites="Poisot2021GloKno">(Poisot <em>et al.</em> 2021)</span>, largely because species interactions are hard to sample <span class="citation" data-cites="Jordano2016SamNet">(Jordano 2016)</span>. Over the past few decades biodiversity data has become increasingly available through remotely collected data and adoption of open data practices <span class="citation" data-cites="Stephenson2020TecAdv Kenall2014OpeFut">(Kenall <em>et al.</em> 2014; Stephenson 2020)</span>. Still, interaction data remains relatively scarce because sampling typically requires human observation. This induces a constraint on the amount, spatial scale, and temporal frequency of resulting data that it is feasible to collect by humans. Many crowdsourced methods for biodiversity data aggregation (e.g. GBIF, eBird) still rely on automated identification of species, which does not easily generalize to interaction sampling. There is interest in using remote methods for interaction sampling, which primarily detect co-occurrence and derive properties like species avoidance from this data <span class="citation" data-cites="Niedballa2019AssAna">(Niedballa <em>et al.</em> 2019)</span>. However, co-occurrence itself is not necessarily indicative of an interaction <span class="citation" data-cites="Blanchet2020CooNot">(Blanchet <em>et al.</em> 2020)</span>. This is an example of semantic confusion around the word “interaction”—for example one might consider competition a type of species interaction, even though it is marked by a lack of co-occurrence between species, unlike other types of interactions, like predation or parasitism, which require both species to be together at the same place and time. Here we consider interaction in the latter sense, where two species have fitness consequences on one-another if (and only if) they are in the sample place at the same time. In addition, here we only consider direct (not higher-order) interactions.</p>
            <p>We cannot feasibly observe all (or even most) of the interactions that occur in an ecosystem. This means we can be confident two species actually interact if we have a record of it (assuming they are correctly identified), but not at all confident that a pair of species <em>do not</em> interact if we have <em>no record</em> of those species observed together. In other words, it is difficult to distinguish <em>true-negatives</em> (two species never interact) from <em>false-negatives</em> (two species interact sometimes, but we do not have a record of this interaction). For a concrete example of a false-negative in a food web, see fig. 1. Because even the most highly sampled systems will still contain false-negatives, there is increasing interest in combining species-level data (e.g. traits, abundance, range, phylogenetic relatedness, etc.) to build models to predict interactions between species we haven’t observed together before <span class="citation" data-cites="Strydom2021RoaPre">(Strydom <em>et al.</em> 2021)</span>. However, the noise of false-negatives could impact the efficacy of our predictive models and have practical consequences for answering questions about interactions <span class="citation" data-cites="deAguiar2019RevBia">(de Aguiar <em>et al.</em> 2019)</span>. This data constraint is amplified as the interaction data we have is geographically biased toward the usual suspects <span class="citation" data-cites="Poisot2021GloKno">(Poisot <em>et al.</em> 2021)</span>. We therefore need a statistical approach to assessing these biases in the observation process and their consequences for our understanding of interaction networks.</p>
            <figure>
            <img src="./figures/concept.png" id="fig:concept" alt="Figure 1: This conceptual example considers a sample of the trophic community of bears, wolves, salmon (pink fish), pike (yellow fish), berry trees, and aspen trees. The true metaweb (all realized interactions across the entire spatial extent) is shown on the left. In the center is what a hypothetical ecologist samples at each site. Notice that although bears are observed co-occurring with both salmon and pike, there was never a direct observation of bears eating pike, even though they actually do. Therefore, this interaction between bears and pike is a false-negative." /><figcaption aria-hidden="true">Figure 1: This conceptual example considers a sample of the trophic community of bears, wolves, salmon (pink fish), pike (yellow fish), berry trees, and aspen trees. The true metaweb (all realized interactions across the entire spatial extent) is shown on the left. In the center is what a hypothetical ecologist samples at each site. Notice that although bears are observed co-occurring with both salmon and pike, there was never a direct observation of bears eating pike, even though they actually do. Therefore, this interaction between bears and pike is a false-negative.</figcaption>
            </figure>
            <p>The importance of <em>sampling effort</em> and its impact on resulting ecological data has produced a rich body of literature. The recorded number of species in a dataset or sample depends on the total number of observations <span class="citation" data-cites="Willott2001SpeAcc Walther1995SamEff">(Walther <em>et al.</em> 1995; Willott 2001)</span>, as do estimates of population abundance <span class="citation" data-cites="Griffiths1998SamEff">(Griffiths 1998)</span>. This relationship between sampling effort, spatial coverage, and species detectability has motivated more quantitatively robust approaches to account for error in sampling data in many contexts: to determine if a given species is extinct <span class="citation" data-cites="Boakes2015InfSpe">(Boakes <em>et al.</em> 2015)</span>, to determine sampling design <span class="citation" data-cites="Moore2016OptEco">(Moore &amp; McCarthy 2016)</span>, and to measure species richness across large scales <span class="citation" data-cites="Carlson2020WhaWou">(Carlson <em>et al.</em> 2020)</span>. In the context of interactions, an initial concern was the compounding effects of limited sampling effort combined with the amalgamation of data (across both study sites, time of year, and taxonomic scales) could lead any empirical set of observations to inadequately reflect the reality of how species interact <span class="citation" data-cites="Paine1988RoaMap">(Paine 1988)</span> or the structure of the network as a whole <span class="citation" data-cites="Martinez1999EffSam McLeod2021SamAsy">(Martinez <em>et al.</em> 1999; McLeod <em>et al.</em> 2021)</span>. <span class="citation" data-cites="Martinez1999EffSam">Martinez <em>et al.</em> (1999)</span> showed that in a plant-endophyte trophic network, network connectance is robust to sampling effort, but this was done in the context of a system for which observation of 62,000 total interactions derived from 164,000 plant-stems was feasible. In some systems (e.g. megafauna food-webs) this many observations is either impractical or infeasible due to the absolute abundance of the species in question.</p>
            <p>The intrinsic properties of ecological communities create several challenges for sampling: first, species are not observed with equal probability—we are much more likely to observe a species of high abundance than one of very low abundance <span class="citation" data-cites="Poisot2015SpeWhy">(Poisot <em>et al.</em> 2015)</span>. <span class="citation" data-cites="Canard2012EmeStr">Canard <em>et al.</em> (2012)</span> presents a null model of food-web structure where species encounter one-another in proportion to each species’ relative-abundance. This assumes that there are no associations in species co-occurrence due to an interaction (perhaps because this interaction is “important” for both species; <span class="citation" data-cites="Cazelles2016TheSpe">Cazelles <em>et al.</em> (2016)</span>), but in this paper we later show increasing strength of these associations leads to increasing probability of false-negatives in interaction data, and that these positive associations are common in existing network data. Second, observed co-occurrence is often equated with meaningful interaction strength, but this is not necessarily the case <span class="citation" data-cites="Blanchet2020CooNot">(Blanchet <em>et al.</em> 2020)</span>—a true “non-interaction” would require that neither of two species, regardless of whether they co-occur, ever exhibit any meaningful effect on the fitness of the other. So, although co-occurrence is not directly indicative of an interaction, it <em>is</em> a precondition for an interaction.</p>
            <p>Here, we illustrate how our confidence that a pair of species never interacts highly depends on sampling effort. We demonstrate how the realized false-negative-rate of interactions is related to the relative abundance of the species pool, and introduce a method to produce a null estimate of the false-negative-rate given total sampling effort (the total count of all interactions seen among all species-pairs) and a method for including uncertainty into model predictions of interaction probabilities to account for observation error. We then confront these models with data, by showing that positive associations in co-occurrence data can increase the realized number of false-negatives and by showing these positive associations are rampant in network datasets. We conclude by recommending that the simulation of sampling effort and species occurrence can and should be used to help design surveys of species interaction diversity <span class="citation" data-cites="Moore2016OptEco">(Moore &amp; McCarthy 2016)</span>, and by advocating use of null models like those presented here as a tool for both guiding design of surveys of species interactions and for including detection error into predictive models.</p>
            <h1 id="accounting-for-false-negatives-in-species-interactions">Accounting for false-negatives in species interactions</h1>
            <p>In this section, we demonstate how differences in species’ relative-abundance can lead to many false-negatives in interaction data. We also introduce a method for producing a null estimate of the false-negative-rate in datasets via simulation. Because the true false-negative-rate is latent, we can never actually be sure how many false-negatives are in our data. However, here we outline an approach to deal with this fact—first by using simulation to estimate the false-negative-rate for a dataset of a fixed size using neutral models of observation. We then illustrate how to incorporate uncertainty directly into predictions of species interactions to account for observation error based on null estimates of both the false-positive rate (as an <em>a priori</em> estimate of species misidentification probability) and false-negative rate (as generated via the method we introduce).</p>
            <h2 id="how-many-observations-of-a-non-interaction-do-we-need-to-be-confident-its-a-true-negative">How many observations of a non-interaction do we need to be confident it’s a true negative?</h2>
            <p>We start with a naive model of interaction detection: we assume that every interacting pair of species is incorrectly observed as not-interacting with an independent and fixed probability, which we denote <span class="math inline"><em>p</em><sub><em>f</em><em>n</em></sub></span> and subsequently refer to as the False-Negative-Rate (FNR). If we observe the same species not-interacting <span class="math inline"><em>N</em></span> times, then the probability of a true-negative (denoted <span class="math inline"><em>p</em><sub><em>t</em><em>n</em></sub></span>) is given by <span class="math inline"><em>p</em><sub><em>t</em><em>n</em></sub> = 1 − (<em>p</em><sub><em>f</em><em>n</em></sub>)<sup><em>N</em></sup></span>. This relation (the cumalitive-distribution-function of geometric distribution, a special case of the negative-binomial distribution) is shown in fig. 2(a) for varying values of <span class="math inline"><em>p</em><sub><em>f</em><em>n</em></sub></span> and illustrates a fundamental link between our ability to reliably say an interaction doesn’t exist—<span class="math inline"><em>p</em><sub><em>t</em><em>n</em></sub></span>—and the number of times <span class="math inline"><em>N</em></span> we have observed a given species. In addition, note that there is no non-zero <span class="math inline"><em>p</em><sub><em>f</em><em>n</em></sub></span> for which we can ever <em>prove</em> that an interaction does not exist—no matter how many observations of non-interactions <span class="math inline"><em>N</em></span> we have, <span class="math inline"><em>p</em><sub><em>t</em><em>n</em></sub> &lt; 1</span>.</p>
            <p>From fig. 2(a) it is clear that the more often we see two species co-occurring, but <em>not interacting</em>, the more likely the interaction is a true-negative. This has several practical consequences: first it means negatives taken outside the overlap of the range of each species aren’t informative because co-occurrence was not possible, and therefore neither was an interaction. In the next section we demonstrate that the distribution of abundance in ecosystems can lead to very high realized values of FNR (<span class="math inline"><em>p</em><sub><em>f</em><em>n</em></sub></span>) simply as an artifact of sampling effort. Second, we can use this relation to compute the expected number of total observations needed to obtain a “goal” number of observations of a particular pair of species (fig. 2(b)). As an example, if we hypothesize that <span class="math inline"><em>A</em></span> and <span class="math inline"><em>B</em></span> do not interact, and we want to see species <span class="math inline"><em>A</em></span> and <span class="math inline"><em>B</em></span> both co-occurring and <em>not interacting</em> 10 times to be confident this is a true negative, then we need an expected 1000 observations of all species if the relative abundances of <span class="math inline"><em>A</em></span> and <span class="math inline"><em>B</em></span> are both <span class="math inline">0.1</span>.</p>
            <figure>
            <img src="./figures/fig1.png" id="fig:geometric" alt="Figure 2: (a) The probability that an observed interaction is a true negative (y-axis) given how many times it has been sampled as a non-interaction (x-axis). Each color reflects a different value of p_{fn}, the false-negative-rate (FNR)—this is effectively the cumulative distribution function (cdf) of the geometric distribution. (b) The expected number of total observations needed (colors) to observe 10 co-occurrences between a species with relative abundance P(A) (x-axis) and a second species with relative abundance P(Y). (c): false-negative-rate (y-axis) as a function of total sampling effort (x-axis) and network size, computed using the method described above. For 500 independent draws from the niche model (Williams &amp; Martinez (2000)) at varying levels of species richness (colors) with connectance drawn according to the flexible-links model (MacDonald et al. (2020)) as described in the main text. For each draw from the niche model, 200 sets of 1500 observations are simulated, for which the mean false-negative-rate at each observation-step is computed. Means denoted with points, with 1 in the first shade and 2 in the second. (d): Same as (c), except using empirical food webs from Mangal database, where richness. The outlier on (d) is a 714 species food-web." /><figcaption aria-hidden="true">Figure 2: <strong>(a)</strong> The probability that an observed interaction is a true negative (y-axis) given how many times it has been sampled as a non-interaction (x-axis). Each color reflects a different value of <span class="math inline"><em>p</em><sub><em>f</em><em>n</em></sub></span>, the false-negative-rate (FNR)—this is effectively the cumulative distribution function (cdf) of the geometric distribution. <strong>(b)</strong> The expected number of total observations needed (colors) to observe 10 co-occurrences between a species with relative abundance <span class="math inline"><em>P</em>(<em>A</em>)</span> (x-axis) and a second species with relative abundance <span class="math inline"><em>P</em>(<em>Y</em>)</span>. <strong>(c)</strong>: false-negative-rate (y-axis) as a function of total sampling effort (x-axis) and network size, computed using the method described above. For 500 independent draws from the niche model (<span class="citation" data-cites="Williams2000SimRul">Williams &amp; Martinez (2000)</span>) at varying levels of species richness (colors) with connectance drawn according to the flexible-links model (<span class="citation" data-cites="MacDonald2020RevLin">MacDonald <em>et al.</em> (2020)</span>) as described in the main text. For each draw from the niche model, 200 sets of 1500 observations are simulated, for which the mean false-negative-rate at each observation-step is computed. Means denoted with points, with 1 in the first shade and 2 in the second. <strong>(d)</strong>: Same as <strong>(c)</strong>, except using empirical food webs from Mangal database, where richness. The outlier on <strong>(d)</strong> is a 714 species food-web.</figcaption>
            </figure>
            <h2 id="false-negatives-as-a-product-of-relative-abundance">False-negatives as a product of relative abundance</h2>
            <p>We now show that the realized FNR changes drastically with sampling effort due to the intrinsic variation of the abundance of individuals of each species within a community. We do this by simulating the process of observation of species interactions, applied both to 243 empirical food webs from the Mangal database <span class="citation" data-cites="Banville2021ManJl">(Banville <em>et al.</em> 2021)</span> and random food-webs generated using the niche model, a simple generative model of food-web structure that accounts for allometric scaling <span class="citation" data-cites="Williams2000SimRul">(Williams &amp; Martinez 2000)</span>. Our neutral model of observation assumes each observed species is drawn in proportion to each species’ abundance at that place and time. The abundance distribution of a community can be reasonably-well described by a log-normal distribution <span class="citation" data-cites="Volkov2003NeuThe">(Volkov <em>et al.</em> 2003)</span>. In addition to the log-normal distribution, we also tested the case where the abundance distribution is derived from power-law scaling <span class="math inline"><em>Z</em><sup>(<em>l</em><em>o</em><em>g</em>(<em>T</em><sub><em>i</em></sub>) − 1)</sup></span> where <span class="math inline"><em>T</em><sub><em>i</em></sub></span> is the trophic level of species <span class="math inline"><em>i</em></span> and <span class="math inline"><em>Z</em></span> is a scaling coefficient <span class="citation" data-cites="Savage2004EffBod">(Savage <em>et al.</em> 2004)</span>, which yields the same qualitative behavior. The practical consequence of abundance distributions spanning many orders of magnitude is that observing two “rare” species interacting requires two low probability events: observing two rare species <em>at the same time</em>.</p>
            <p>To simulate the process of observation, for an ecological network <span class="math inline"><em>M</em></span> with <span class="math inline"><em>S</em></span> species, we sample relative abundances for each species from a standard-log-normal distribution. For each true interaction in the adjacency matrix <span class="math inline"><em>M</em></span> (i.e. <span class="math inline"><em>M</em><sub><em>i</em><em>j</em></sub> = 1</span>) we estimate the probability of observing both species <span class="math inline"><em>i</em></span> and <span class="math inline"><em>j</em></span> at a given place and time by simulating <span class="math inline"><em>n</em></span> observations of all individuals of any species, where the species of the individual observed at the <span class="math inline">{1, 2, …, <em>n</em>}</span>-th observation is drawn from the generated categorical distribution of abundances. For each pair of species <span class="math inline">(<em>i</em>, <em>j</em>)</span>, if both <span class="math inline"><em>i</em></span> and <span class="math inline"><em>j</em></span> are observed within the n-observations, the interaction is tallied as a true positive if <span class="math inline"><em>M</em><sub><em>i</em><em>j</em></sub> = 1</span>. If only one of <span class="math inline"><em>i</em></span> or <span class="math inline"><em>j</em></span> are observed—but not both—in these <span class="math inline"><em>n</em></span> observations, but <span class="math inline"><em>M</em><sub><em>i</em><em>j</em></sub> = 1</span>, this is counted as a false-negative, and a true-negative otherwise. For each pair of species <span class="math inline">(<em>i</em>, <em>j</em>)</span>, if both <span class="math inline"><em>i</em></span> and <span class="math inline"><em>j</em></span> are observed within the n-observations, the interaction is tallied as a true positive if <span class="math inline"><em>M</em><sub><em>i</em><em>j</em></sub> = 1</span>. If only one of <span class="math inline"><em>i</em></span> or <span class="math inline"><em>j</em></span> are observed—but not both—in these <span class="math inline"><em>n</em></span> observations, but <span class="math inline"><em>M</em><sub><em>i</em><em>j</em></sub> = 1</span>, this is counted as a false-negative, and a true-negative otherwise (<span class="math inline"><em>M</em><sub><em>i</em><em>j</em></sub> = 0</span>). This process is illustrated conceptually in fig. 3(a).</p>
            <p>In fig. 2(c) we see this model of observation applied to niche model networks across varying levels of species richness, and in fig. 2(d) the observation model applied to Mangal food webs. For all niche model simulations in this manuscript, for a given number of species <span class="math inline"><em>S</em></span> the number of interactions is drawn from the flexible-links model fit to Mangal data <span class="citation" data-cites="MacDonald2020RevLin">(MacDonald <em>et al.</em> 2020)</span>, effectively drawing the number of interactions <span class="math inline"><em>L</em></span> for a random niche model food-web as</p>
            <p><span class="math display"><em>L</em> ∼ BetaBinomial(<em>S</em><sup>2</sup> − <em>S</em> + 1, <em>μ</em><em>ϕ</em>, 1 − <em>μ</em><em>ϕ</em>)</span></p>
            <p>where the maximum <em>a posteriori</em> (MAP) estimate of <span class="math inline">(<em>μ</em>, <em>ϕ</em>)</span> applied to Mangal data from <span class="citation" data-cites="MacDonald2020RevLin">(MacDonald <em>et al.</em> 2020)</span> is <span class="math inline">(<em>μ</em> = 0.086, <em>ϕ</em> = 24.3)</span>. All simulations were done with 500 independent replicates of unique niche model networks per unique number of total interactions observed <span class="math inline"><em>n</em></span>. All analyses presented here are done in Julia v1.8 <span class="citation" data-cites="Bezanson2015JulFre">(Bezanson <em>et al.</em> 2015)</span> using both EcologicalNetworks.jl v0.5 and Mangal.jl v0.4 <span class="citation" data-cites="Banville2021ManJl">(Banville <em>et al.</em> 2021)</span> and are hosted on  (<strong>link removed for double-blind review</strong>). Note that the empirical data, for the reasons described above, very likely already contains many false-negatives, we’ll revisit this issue in the final section.</p>
            <p>From fig. 2(c) it is evident that the number of species considered in a study is inseparable from the false-negative-rate in that study, and this effect should be taken into account when designing samples of ecological networks in the future. We see a similar qualitative pattern in empirical networks (fig. 2(d)) where the FNR drops off quickly as a function of observation effort, mediated by total richness. The practical consequence of the bottom row of fig. 2 when conducting an analysis is whether there are enough total number of observed interactions (the x-axis) for the threshold FNR we deem acceptable (the y-axis) is feasible. This raises two points: first, empirical data on interactions are subject to the practical limitations of funding and human-work hours, and therefore existing data tend to fall on the order of hundreds or thousands observations of individuals per site. Clear aggregation of data on sampling effort has proven difficult to find and a meta-analysis of network data and sampling effort seems both pertinent and necessary, in addition to the effects of aggregation of interactions across taxonomic scales <span class="citation" data-cites="Gauzens2013FooAgg Giacomuzzo2021FooWeb">(Gauzens <em>et al.</em> 2013; Giacomuzzo &amp; Jordán 2021)</span>. This inherent limitation on in-situ sampling means we should optimize where we sample across space so that for a given number of samples, we obtain the maximum information possible. Second, what is meant by “acceptable” FNR? This raises the question: does a shifting FNR lead to rapid transitions in our ability inference and predictions about the structure and dynamics of networks, or does it produce a roughly linear decay in model efficacy? We explore this in the final section.</p>
            <p>We conclude this section by advocating for the use of neutral models similar to above to generate expectations about the number of false-negatives in a dataset of a given size. This could prove fruitful both for designing surveys of interactions but also because we may want to incorporate models of imperfect detection error into predictive interactions models, as <span class="citation" data-cites="Joseph2020NeuHie">Joseph (2020)</span> does for species occurrence modeling. Additionally, we emphasize that one must consider the context for sampling—is the goal to detect a particular species (as in fig. 2(c)), or to get a representative sample of interactions across the species pool? These arguments are well-considered when sampling individual species <span class="citation" data-cites="Willott2001SpeAcc">(Willott 2001)</span>, but have not yet been adopted for designing samples of communities.</p>
            <h2 id="including-observation-error-in-interaction-predictions">Including observation error in interaction predictions</h2>
            <p>Here we show how to incorporate uncertainty into model predictions of interaction probability to account for imperfect observation (both false-negatives and false-positives). Models for interaction prediction typically yield a probability of interaction between each pair of species, <span class="math inline"><em>p</em><sub><em>i</em><em>j</em></sub></span>. When these are considered with uncertainty, it is usually model-uncertainty, e.g. the variance in the interaction probability prediction across several cross-validation folds, where the data is split into training and test sets several times. The method we introduce adjusts the value of a model’s predictions to produce a distribution of interaction probabilities corrected by a given false-negative-rate <span class="math inline"><em>p</em><sub><em>f</em><em>n</em></sub></span> and false-positive-rate <span class="math inline"><em>p</em><sub><em>f</em><em>p</em></sub></span> (outlined in figure fig. 3). First we describe how to sample from this distribution of adjusted interaction probabilities via simulation, and show that this distribution can be well-approximated analytically.</p>
            <figure>
            <img src="./figures/uncertainty_sampler.png" id="fig:resampling_concept" alt="Figure 3: (a) The process for estimating the false-negative-rate (FNR) for an interaction dataset consisting of N total observed interactions. (b) The method for resampling interaction probability based on estimates of false-negative and false-positive rates. (c) The method for interaction probability resampling applied to three mammals and three parasites from the Hadfield et al. (2014) dataset. The original probability p_{ij} is indicated with a vertical dashed line. The histogram is simulated from the resampling process, and the line indicates the gaussian approximation to this distribution. Both resampling simulations and the gaussian approximation is applied with n_p =150" /><figcaption aria-hidden="true">Figure 3: (a) The process for estimating the false-negative-rate (FNR) for an interaction dataset consisting of <span class="math inline"><em>N</em></span> total observed interactions. (b) The method for resampling interaction probability based on estimates of false-negative and false-positive rates. (c) The method for interaction probability resampling applied to three mammals and three parasites from the <span class="citation" data-cites="Hadfield2014TalTwo">Hadfield <em>et al.</em> (2014)</span> dataset. The original probability <span class="math inline"><em>p</em><sub><em>i</em><em>j</em></sub></span> is indicated with a vertical dashed line. The histogram is simulated from the resampling process, and the line indicates the gaussian approximation to this distribution. Both resampling simulations and the gaussian approximation is applied with <span class="math inline"><em>n</em><sub><em>p</em></sub> = 150</span></figcaption>
            </figure>
            <p>To get an estimate of each interaction probability that accounts for observation error, we resample the output prediction from an arbitrary model for each interaction <span class="math inline"><em>p</em><sub><em>i</em><em>j</em></sub></span> by simulating a set of several ‘particles.’ Each particle is a realization of an interaction <em>actually occurring</em> assuming <span class="math inline"><em>p</em><sub><em>i</em><em>j</em></sub></span> is a correct estimate of the probability of an interaction being <em>observed</em>. Each particle starts as being drawn as true or false according to <span class="math inline"><em>p</em><sub><em>i</em><em>j</em></sub></span>, and then adjusting this by the rate of observation error given by <span class="math inline"><em>p</em><sub><em>f</em><em>p</em></sub></span> and <span class="math inline"><em>p</em><sub><em>f</em><em>n</em></sub></span> to yield a single boolean outcome for each particle (“Resampling” within fig. 3(b)). Across of many particles, the resulting frequency of ‘true’ outcomes is a single resample of the probability <span class="math inline"><em>p</em><sub><em>i</em><em>j</em></sub><sup>*</sup></span> that the interaction actually <em>occurred</em>, not just that it was <em>observed</em>. Across several samples each of several particles, this forms a distribution of probabilities which are adjusted by the true and false-negative-rates.</p>
            <p>There is also an analytic way to approximate this distribution using the normal approximation to binomial. As a reminder, as the total number of samples <span class="math inline"><em>N</em></span> from a binomial distribution for <span class="math inline"><em>n</em></span> trials with success probability <span class="math inline"><em>p</em></span> from approaches infinity, the sum of total successes across all samples approaches a normal distribution with mean <span class="math inline"><em>n</em><em>p</em></span> and variance <span class="math inline"><em>n</em><em>p</em>(1 − <em>p</em>)</span>. For notation, here we refer to a normal distribution with mean <span class="math inline"><em>μ</em></span> and standard-deviation <span class="math inline"><em>σ</em></span> as <span class="math inline">𝒩(<em>μ</em>, <em>σ</em>)</span>. We can use this to correct the estimate <span class="math inline"><em>p</em><sub><em>i</em><em>j</em></sub></span> based on the expected false-negative-rate <span class="math inline"><em>p</em><sub><em>f</em><em>n</em></sub></span> and false-positive rate <span class="math inline"><em>p</em><sub><em>f</em><em>p</em></sub></span> to obtain the limiting distribution as the number of resamples approaches infinity for the resampled <span class="math inline"><em>p</em><sub><em>i</em><em>j</em></sub><sup>*</sup></span> for a given number of particles <span class="math inline"><em>n</em><sub><em>p</em></sub></span>. We do this by first adjusting for the rates of observation error to get the mean resampled probability, <span class="math inline">𝔼[<em>p</em><sub><em>i</em><em>j</em></sub><sup>*</sup>]</span>, as</p>
            <p><span class="math display">𝔼[<em>p</em><sub><em>i</em><em>j</em></sub><sup>*</sup>] = <em>p</em><sub><em>i</em><em>j</em></sub>(1 − <em>p</em><sub><em>f</em><em>p</em></sub>) + (1 − <em>p</em><sub><em>i</em><em>j</em></sub>)<em>p</em><sub><em>f</em><em>n</em></sub></span></p>
            <p>This yields the normal approximation</p>
            <p><span class="math display">$$
            \sum_{i=1}^{n_p} p_{ij}^* \sim \mathcal{N}\bigg(n_p \cdot \mathbb{E}[p_{ij}^*], \sqrt{n_p\mathbb{E}[p_{ij}^*] (1- \mathbb{E}[p_{ij}^*])}\bigg)
            $$</span></p>
            <p>which then can be converted back to a distribution of frequency of successes to yield the final approximation</p>
            <p><span id="eq:eq1"><span class="math display">$$p_{ij}^* \sim \mathcal{N}\bigg( \mathbb{E}[p_{ij}^*] , \sqrt{\frac{\mathbb{E}[p_{ij}^*]
            (1-\mathbb{E}[p_{ij}^*] )}{n_p}} \bigg)\qquad(1)$$</span></span></p>
            <p>We can then further truncate this distribution to remain on the interval <span class="math inline">(0, 1)</span>, as the output is a probability, although in practice often the probability mass outside <span class="math inline">(0, 1)</span> is extremely low except for <span class="math inline"><em>p</em><sub><em>i</em><em>j</em></sub></span> values very close to 0 or 1. As an example case study, we use a boosted-regression-tree to predict interactions in a host-parasite network <span class="citation" data-cites="Hadfield2014TalTwo">(Hadfield <em>et al.</em> 2014)</span> (with features derived in the same manner as <span class="citation" data-cites="Strydom2021RoaPre">Strydom <em>et al.</em> (2021)</span> derives features on this data) to produce a set of interaction predictions. We then applied this method to a set of a few resampled interaction probabilities between mammals and parasite species shown in figure fig. 3(c).</p>
            <p>Why is this useful? For one, this analytic method avoids the extra computation required by simulating samples from this distribution directly. Further, it enables the extension of the natural analogue between <span class="math inline"><em>n</em><sub><em>p</em></sub></span> (the number of particles) and the number of observations of co-occurrence for a given pair of species—the fewer the particles, the higher the variance of the resulting approximation. The normal approximation is undefined for 0 particles (i.e. 0 observations co-occurrence), although as <span class="math inline"><em>n</em><sub><em>p</em></sub></span> approaches 0 the approximated normal (once truncated) approaches the uniform distribution on the interval <span class="math inline">(0, 1)</span>, the maximum entropy distribution where we have no information about the possibility of an interaction.</p>
            <p>This also has implications for what we mean by ‘uncertainty’ in interaction predictions. A model’s prediction can be ‘uncertain’ in two different ways: (1) the model’s predictions may have high variance, or (2) the model’s predictions may be centered around a probability of interaction of <span class="math inline">0.5</span>, where we are the most unsure about whether this interaction exists. Improving the incorporation of different forms of uncertainty in probabilistic interaction predictions seems a necessary next step toward understanding what pairs of species we know the least about, in order to prioritize sampling to provide the most new information possible.</p>
            <h1 id="positive-associations-in-co-occurrence-increase-the-false-negative-rate">Positive associations in co-occurrence increase the false-negative-rate</h1>
            <p>The model above doesn’t consider the possibility that there are positive or negative associations which shift the probability of species cooccurrence away from what is expected based on their relative abundances due to their interaction <span class="citation" data-cites="Cazelles2016TheSpe">(Cazelles <em>et al.</em> 2016)</span>. However, here we demonstrate that the probability of having a false-negative can be higher if there is some positive association in the occurrence of species <span class="math inline"><em>A</em></span> and <span class="math inline"><em>B</em></span>. If we denote the probability that we observe the co-occurrence of two species <span class="math inline"><em>A</em></span> and <span class="math inline"><em>B</em></span> as <span class="math inline"><em>P</em>(<em>A</em><em>B</em>)</span> and if there is no association between the marginal probabilities of observing <span class="math inline"><em>A</em></span> and observing <span class="math inline"><em>B</em></span>, denoted <span class="math inline"><em>P</em>(<em>A</em>)</span> and <span class="math inline"><em>P</em>(<em>B</em>)</span> respectively, then the probability of observing their co-occurrence is the product of the marginal probabilities for each species, <span class="math inline"><em>P</em>(<em>A</em><em>B</em>) = <em>P</em>(<em>A</em>)<em>P</em>(<em>B</em>)</span>. In the other case where there is some positive strength of association between observing both <span class="math inline"><em>A</em></span> and <span class="math inline"><em>B</em></span> because this interaction is “important” for each species, then the probability of observation both <span class="math inline"><em>A</em></span> and <span class="math inline"><em>B</em></span>, <span class="math inline"><em>P</em>(<em>A</em><em>B</em>)</span>, is greater than <span class="math inline"><em>P</em>(<em>A</em>)<em>P</em>(<em>B</em>)</span> as <span class="math inline"><em>P</em>(<em>A</em>)</span> and <span class="math inline"><em>P</em>(<em>B</em>)</span> are not independent and instead are positively correlated, i.e. <span class="math inline"><em>P</em>(<em>A</em><em>B</em>) &gt; <em>P</em>(<em>A</em>)<em>P</em>(<em>B</em>)</span>. In this case, the probability of observing a single false-negative in our naive model from fig. 2(a) is <span class="math inline"><em>p</em><sub><em>f</em><em>n</em></sub> = 1 − <em>P</em>(<em>A</em><em>B</em>)</span>, which due to the above inequality implies <span class="math inline"><em>p</em><sub><em>f</em><em>n</em></sub> &gt; 1 − <em>P</em>(<em>A</em>)<em>P</em>(<em>B</em>)</span>. This indicates an increasingly greater probability of a false negative as the strength of association gets stronger, <span class="math inline"><em>P</em>(<em>A</em><em>B</em>) → <em>P</em>(<em>A</em><em>B</em>) ≫ <em>P</em>(<em>A</em>)<em>P</em>(<em>B</em>)</span>. However, this still does not consider variation in species abundance in space and time <span class="citation" data-cites="Poisot2015SpeWhy">(Poisot <em>et al.</em> 2015)</span>. If positive or negative associations between species structure variation in the distribution of <span class="math inline"><em>P</em>(<em>A</em><em>B</em>)</span> across space/time, then the spatial/temporal biases induced by data collection would further impact the realized false-negative-rate, as the probability of false negative would not be constant for each pair of species across sites.</p>
            <p>To test for these positive associations in data we scoured Mangal for datasets with many spatial or temporal replicates of the same system, which led the the resulting seven datasets set in figure fig. 4. For each dataset, we compute the marginal probability <span class="math inline"><em>P</em>(<em>A</em>)</span> of occurrence of each species <span class="math inline"><em>A</em></span> across all networks in the dataset. For each pair of interacting species <span class="math inline"><em>A</em></span> and <span class="math inline"><em>B</em></span>, we then compute and compare the probability of co-occurrence if each species occurs independently, <span class="math inline"><em>P</em>(<em>A</em>)<em>P</em>(<em>B</em>)</span>, to the empirical joint probability of co-occurrence, <span class="math inline"><em>P</em>(<em>A</em><em>B</em>)</span>. Following our analysis above, if <span class="math inline"><em>P</em>(<em>A</em><em>B</em>)</span> is greater than <span class="math inline"><em>P</em>(<em>A</em>)<em>P</em>(<em>B</em>)</span>, then we expect our neutral estimates of the FNR above to underestimate the realized FNR. In fig. 4, we see the difference between <span class="math inline"><em>P</em>(<em>A</em><em>B</em>)</span> and <span class="math inline"><em>P</em>(<em>A</em>)<em>P</em>(<em>B</em>)</span> for the seven suitable datasets with enough spatio-temporal replicates and a shared taxonomic backbone (meaning all individual networks use common species identifiers) found on Mangal to perform this analysis. Further details about each dataset are reported in tbl. 1.</p>
            <p>In each of these datasets, the joint probability of co-occurrence <span class="math inline"><em>P</em>(<em>A</em><em>B</em>)</span> is decisively greater than our expectation if species co-occur in proportion to their relative abundance <span class="math inline"><em>P</em>(<em>A</em>)<em>P</em>(<em>B</em>)</span>. This suggests that there may not be as many “neutrally forbidden links” <span class="citation" data-cites="Canard2012EmeStr">(Canard <em>et al.</em> 2012)</span> as we might think, and that the reason we do not have records of interactions between rare species is probably due to observation error. This has serious ramifications for the widely observed property of nestedness seen in bipartite networks <span class="citation" data-cites="Bascompte2007PlaMut">(Bascompte &amp; Jordano 2007)</span>—perhaps the reason we have lots of observations between generalists is because they are more abundant, and this is particularly relevant as we have strong evidence that generalism drives abundance <span class="citation" data-cites="Song2022GenDri">(Song <em>et al.</em> 2022a)</span>, not vice-versa.</p>
            <figure>
            <img src="./figures/fig2.png" id="fig:mangal" alt="Figure 4: The difference between joint-probability of co-occurrence (P(AB)) and expected probability of co-occurrence under independence (P(A)P(B)) for interacting species for each dataset. The red-dashed line indicates 0 (no association). Each histogram represents a density, meaning the area of the entire curve sums to 1. The continuous density estimate (computed using local smoothing) is shown in grey. The p-value on each plot is the result of a one-sided t-test comparing the mean of each distribution to 0." /><figcaption aria-hidden="true">Figure 4: The difference between joint-probability of co-occurrence (<span class="math inline"><em>P</em>(<em>A</em><em>B</em>)</span>) and expected probability of co-occurrence under independence (<span class="math inline"><em>P</em>(<em>A</em>)<em>P</em>(<em>B</em>)</span>) for interacting species for each dataset. The red-dashed line indicates 0 (no association). Each histogram represents a density, meaning the area of the entire curve sums to 1. The continuous density estimate (computed using local smoothing) is shown in grey. The p-value on each plot is the result of a one-sided t-test comparing the mean of each distribution to 0.</figcaption>
            </figure>
            <div id="tbl:id">
            <table>
            <caption>Table 1: The datasets used in the above analysis (Fig 2). The table reports the type of each dataset, the total number of networks in each dataset <span class="math inline">(<em>N</em>)</span>, the total species richness in each dataset <span class="math inline">(<em>S</em>)</span>, the connectance of each metaweb (all interactions across the entire spatial-temporal extent) <span class="math inline">(<em>C</em>)</span>, the mean species richness across each local network <span class="math inline"><em>S̄</em></span>, the mean connectance of each local network <span class="math inline"><em>C̄</em></span>, the mean <span class="math inline"><em>β</em></span>-diversity among overlapping species across all pairs of network species (<span class="math inline"><em>β̄</em><sub><em>O</em><em>S</em></sub></span>), and the mean <span class="math inline"><em>β</em></span>-diversity among all species in the metaweb (<span class="math inline"><em>β̄</em><sub><em>W</em><em>N</em></sub></span>). Both metrics are computed using KGL <span class="math inline"><em>β</em></span>-diversity <span class="citation" data-cites="Koleff2003MeaBet">(Koleff <em>et al.</em> 2003)</span></caption>
            <thead>
            <tr class="header">
            <th style="text-align: center;">Network</th>
            <th style="text-align: center;">Type</th>
            <th style="text-align: center;"><span class="math inline"><em>N</em></span></th>
            <th style="text-align: center;"><span class="math inline"><em>S</em></span></th>
            <th style="text-align: center;"><span class="math inline"><em>C</em></span></th>
            <th style="text-align: center;"><span class="math inline"><em>S̄</em></span></th>
            <th style="text-align: center;"><span class="math inline"><em>C̄</em></span></th>
            <th style="text-align: center;"><span class="math inline"><em>β̄</em><sub><em>O</em><em>S</em></sub></span></th>
            <th style="text-align: center;"><span class="math inline"><em>β̄</em><sub><em>W</em><em>N</em></sub></span></th>
            </tr>
            </thead>
            <tbody>
            <tr class="odd">
            <td style="text-align: center;"><span class="citation" data-cites="Kopelke2017FooStr">Kopelke <em>et al.</em> (2017)</span></td>
            <td style="text-align: center;">Food Web</td>
            <td style="text-align: center;">100</td>
            <td style="text-align: center;">98</td>
            <td style="text-align: center;">0.037</td>
            <td style="text-align: center;">7.87</td>
            <td style="text-align: center;">0.142</td>
            <td style="text-align: center;">1.383</td>
            <td style="text-align: center;">1.972</td>
            </tr>
            <tr class="even">
            <td style="text-align: center;"><span class="citation" data-cites="Thompson2000ResSol">Thompson &amp; Townsend (2000)</span></td>
            <td style="text-align: center;">Food Web</td>
            <td style="text-align: center;">18</td>
            <td style="text-align: center;">566</td>
            <td style="text-align: center;">0.014</td>
            <td style="text-align: center;">80.67</td>
            <td style="text-align: center;">0.049</td>
            <td style="text-align: center;">1.617</td>
            <td style="text-align: center;">1.594</td>
            </tr>
            <tr class="odd">
            <td style="text-align: center;"><span class="citation" data-cites="Havens1992ScaStr">Havens (1992)</span></td>
            <td style="text-align: center;">Food Web</td>
            <td style="text-align: center;">50</td>
            <td style="text-align: center;">188</td>
            <td style="text-align: center;">0.065</td>
            <td style="text-align: center;">33.58</td>
            <td style="text-align: center;">0.099</td>
            <td style="text-align: center;">1.468</td>
            <td style="text-align: center;">1.881</td>
            </tr>
            <tr class="even">
            <td style="text-align: center;"><span class="citation" data-cites="Ponisio2017OppAtt">Ponisio <em>et al.</em> (2017)</span></td>
            <td style="text-align: center;">Pollinator</td>
            <td style="text-align: center;">100</td>
            <td style="text-align: center;">226</td>
            <td style="text-align: center;">0.079</td>
            <td style="text-align: center;">23.0</td>
            <td style="text-align: center;">0.056</td>
            <td style="text-align: center;">1.436</td>
            <td style="text-align: center;">1.870</td>
            </tr>
            <tr class="odd">
            <td style="text-align: center;"><span class="citation" data-cites="Hadfield2014TalTwo">Hadfield <em>et al.</em> (2014)</span></td>
            <td style="text-align: center;">Host-Parasite</td>
            <td style="text-align: center;">51</td>
            <td style="text-align: center;">327</td>
            <td style="text-align: center;">0.085</td>
            <td style="text-align: center;">32.71</td>
            <td style="text-align: center;">0.337</td>
            <td style="text-align: center;">1.477</td>
            <td style="text-align: center;">1.952</td>
            </tr>
            <tr class="even">
            <td style="text-align: center;"><span class="citation" data-cites="Closs1994SpaTem">Closs &amp; Lake (1994)</span></td>
            <td style="text-align: center;">Food Web</td>
            <td style="text-align: center;">12</td>
            <td style="text-align: center;">61</td>
            <td style="text-align: center;">0.14</td>
            <td style="text-align: center;">29.09</td>
            <td style="text-align: center;">0.080</td>
            <td style="text-align: center;">1.736</td>
            <td style="text-align: center;">1.864</td>
            </tr>
            <tr class="odd">
            <td style="text-align: center;"><span class="citation" data-cites="CaraDonna2017IntRew">CaraDonna <em>et al.</em> (2017)</span></td>
            <td style="text-align: center;">Pollinator</td>
            <td style="text-align: center;">86</td>
            <td style="text-align: center;">122</td>
            <td style="text-align: center;">0.18</td>
            <td style="text-align: center;">21.42</td>
            <td style="text-align: center;">0.312</td>
            <td style="text-align: center;">1.527</td>
            <td style="text-align: center;">1.907</td>
            </tr>
            </tbody>
            </table>
            </div>
            <h1 id="the-impact-of-false-negatives-on-network-properties-and-prediction">The impact of false-negatives on network properties and prediction</h1>
            <p>Here, we assess the effect of false-negatives on our ability to make predictions about interactions, as well as their effect on network structure. The prevalence of false-negatives in data is the catalyst for interaction prediction in the first place, and as a result methods have been proposed to counteract this bias <span class="citation" data-cites="Stock2017LinFil Poisot2022NetEmb">(Stock <em>et al.</em> 2017; Poisot <em>et al.</em> 2022)</span>. However, it is feasible that the FNR in a given dataset is so high that it could induce too much noise for an interaction prediction model to detect the signal of possible interaction between species.</p>
            <p>To test this we use the dataset from <span class="citation" data-cites="Hadfield2014TalTwo">Hadfield <em>et al.</em> (2014)</span> that describes host-parasite interaction networks sampled across 51 sites, and the same method as <span class="citation" data-cites="Strydom2021RoaPre">Strydom <em>et al.</em> (2021)</span> to extract latent features for each species in this dataset based on applying PCA to the co-occurrence matrix. We then predict a metaweb (equivalent to predicting true or false for an interaction between each species pair, effectively a binary classification problem) from these species-level features using four candidate models for binary classification—three often used machine-learning (ML) methods (Boosted Regression Tree (BRT), Random Forest (RF), Decision Tree (DT)), and one naive model from classic statistics (Logistic Regression (LR)). Each of the ML models are bootstrap aggregated (or bagged) with 100 replicates each. We partition the data into 80-20 training-test splits, and then seed the training data with false negatives at varying rates, but crucially do <em>nothing</em> to the test data. We fit all of these models using MLJ.jl, a high-level Julia framework for a wide-variety of ML models <span class="citation" data-cites="Blaom2020MljJul">(Blaom <em>et al.</em> 2020)</span>. We evaluate the efficacy of these models using two common measures of binary classifier performance: the area under the receiver-operator curve (ROC-AUC) and the area under the precision-recall curve (PR-AUC), for more details see <span class="citation" data-cites="Poisot2022GuiPre">Poisot (2022)</span>. Here, PR-AUC is slightly more relevant as it is a better indicator of prediction of false-negatives. The results of these simulations are shown in fig. 5(a &amp; b).</p>
            <figure>
            <img src="./figures/fig3.png" id="fig:addedfnr" alt="Figure 5: (a) The area-under the receiver-operator curve (ROC-AUC) and (b) The area-under the precision-recall curve (PR-AUC; right) for each different predictive model (colors/shapes) across a spectrum of the proportion of added false-negatives (x-axis). (c) The mean trophic-level of all species in a network generated with the niche model across different species richnesses (colors). For each value of the FNR, the mean trophic level was computed across 50 replicates. The shaded region for each line is one standard-deviation across those replicates." /><figcaption aria-hidden="true">Figure 5: <strong>(a)</strong> The area-under the receiver-operator curve (ROC-AUC) and <strong>(b)</strong> The area-under the precision-recall curve (PR-AUC; right) for each different predictive model (colors/shapes) across a spectrum of the proportion of added false-negatives (x-axis). <strong>(c)</strong> The mean trophic-level of all species in a network generated with the niche model across different species richnesses (colors). For each value of the FNR, the mean trophic level was computed across 50 replicates. The shaded region for each line is one standard-deviation across those replicates.</figcaption>
            </figure>
            <p>One interesting result seen in fig. 5(a &amp; b) is that the ROC-AUC value does not approach random in the same way the PR-AUC curve does as we increase the added FNR. The reason for this is that ROC-AUC is fundamentally not as useful a metric in assessing predictive capacity as PR-AUC. As we keep adding more false-negatives, the network eventually becomes a zeros matrix, and these models can still learn to predict “no-interaction” for all possible species pairs, which does far better than random guessing (ROC-AUC = 0.5) in terms of the false positive rate (one of the components of ROC-AUC). This highlights a more broad issue of label class imbalance, meaning there are far more non-interactions than interactions in data. A full treatment of the importance of class-balance is outside the scope of this paper, but is explored in-depth in <span class="citation" data-cites="Poisot2022GuiPre">Poisot (2022)</span>. Further we see, if anything, gradual decline in the performance of the model until we reach very high FNR levels (i.e. <span class="math inline"><em>p</em><sub><em>f</em><em>n</em></sub> &gt; 0.7</span>). This is consistent with other recent work <span class="citation" data-cites="Gupta2023HowMan">(Gupta <em>et al.</em> 2023)</span>, although it must be considered that the empircal data on which these models are trained already are almost certain to already contain false-negatives.</p>
            <p>Although these ML models are surprisingly performant at link prediction given their simplicity, there have been several major developments in applying deep-learning methods to many tasks in network inference and prediction—namely graph-representation learning (GRL, <span class="citation" data-cites="Khoshraftar2022SurGra">Khoshraftar &amp; An (2022)</span>) and graph convolutional networks <span class="citation" data-cites="Zhang2019GraCon">(Zhang <em>et al.</em> 2019)</span>. At this time, these advances can not yet be applied to ecological networks because they require far more data than we currently have. We already have lots of features that could be used as inputs into these models (i.e. species level data about occurrence, genomes, abundance, etc.), but our network datasets barely get into the hundreds of local networks sampled across space and time (tbl. 1). Once we start to get into the thousands, these models will become more useful, but this can only be done with systematic monitoring of interactions. This again highlights the need to optimize our sampling effort to maximize the amount of information contained in our data given the expense of sampling interactions.</p>
            <p>We also consider how the FNR affects network properties. In fig. 5(c) we see the mean trophic level across networks simulated using the niche model (as above), across a spectrum of FNR values. In addition to the clear dependence on richness, we see that mean trophic level, despite varying widely between niche model simulations, tends to be relatively robust to false-negatives and does not deviate widely from the true value until very large FNRs. This is not entirely unsurprising. Removing links randomly from a food-web is effectively the inverse problem of the emergence of a giant component (more than half of the nodes are in a connected network) in random graphs (see <span class="citation" data-cites="Li2021PerCom">Li <em>et al.</em> (2021)</span> for a thorough review). The primary difference being that we are removing edges, not adding them, and thus we are witnessing the dissolution of a giant component, rather than the emergence of one. Further applications of percolation theory <span class="citation" data-cites="Li2021PerCom">(Li <em>et al.</em> 2021)</span> to the topology of sampled ecological networks could improve our understanding of how false-negatives impact the inferences about the structure and dynamics on these networks.</p>
            <h1 id="discussion">Discussion</h1>
            <p>Species interactions enable the persistence and functioning of ecosystems, but our understanding of interactions is limited due to the intrinsic difficulty of sampling them. Here we have provided a null model for the expected number of false-negatives in an interaction dataset. We demonstrated that we expect many false-negatives in species interaction datasets purely due to the intrinsic variation of abundances within a community. We also, for the first time to our knowledge, measured the strength of association between co-occurrence and interactions <span class="citation" data-cites="Cazelles2016TheSpe">(Cazelles <em>et al.</em> 2016)</span> across many empirical systems, and found that these positive associations are both very common, and showed algebraically that they increase the realized FNR. We have also shown that false-negatives could further impact our ability to both predict interactions and infer properties of the networks, which highlights the need for further research into methods for correcting this bias in existing data.</p>
            <p>A better understanding of how false-negatives impact species interaction data is a practical necessity—both for inference of network structure and dynamics, but also for prediction of interactions by using species level information. False-negatives could pose a problem for many forms of inference in network ecology. For example, inferring the dynamic stability of a network could be prone to error if the observed network is not sampled “enough.” What exactly “enough” means is then specific to the application, and should be assessed via methods like those here when designing samples. Further, predictions about network rewiring <span class="citation" data-cites="Thompson2017DisGov">(Thompson &amp; Gonzalez 2017)</span> due to range shifts in response to climate change could be error-prone without accounting for interactions that have not been observed but that still may become climatically infeasible. As is evident from fig. 2(a), we can never guarantee there are no false-negatives in data. In recent years, there has been interest toward explicitly accounting for false-negatives in models <span class="citation" data-cites="Stock2017LinFil Young2021RecPla">(Stock <em>et al.</em> 2017; Young <em>et al.</em> 2021)</span>, and a predictive approach to networks—rather than expecting our samples to fully capture all interactions <span class="citation" data-cites="Strydom2021RoaPre">(Strydom <em>et al.</em> 2021)</span>. As a result, better models for predicting interactions are needed for interaction networks. This includes explicitly accounting for observation error <span class="citation" data-cites="Johnson2021BayEst">(Johnson &amp; Larremore 2021)</span>—certain classes of models have been used to reflect hidden states which account for detection error in occupancy modeling <span class="citation" data-cites="Joseph2020NeuHie">(Joseph 2020)</span>, and could be integrated in the predictive models of interactions in the future.</p>
            <p>This work has several practical consequences for the design of surveys for species’ interactions. Simulating the process of observation could be a powerful tool for estimating the sampling effort required by a study that takes relative abundance into account, and provides a null baseline for expected FNR. It is necessary to take the size of the species pool into account when deciding how many total samples is sufficient for an “acceptable” FNR (fig. 2(c &amp; d)). Further the spatial and temporal turnover of interactions means any approach to sampling prioritization must be spatiotemporal. We demonstrated earlier that observed negatives outside of the range of both species aren’t informative, and therefore using species distribution models could aid in this spatial prioritization of sampling sites.</p>
            <p>We also should address the impact of false-negatives on the inference of process and causality in community ecology. We demonstrated that in model food webs, false-negatives do not impact the measure of total trophic levels until very high FNR (figure fig. 5(c)), although we cannot generalize this further to other properties. This has immediate practical concern for how we design what taxa to sample—does it matter if the sampled network is fully connected? It has been shown that the stability of subnetworks can be used to infer the stability of the metaweb paper beyond a threshold of samples <span class="citation" data-cites="Song2022RapMon">(Song <em>et al.</em> 2022b)</span>. But does this extend to other network properties? And how can we be sure we are at the threshold at which we can be confident our sample characterizes the whole system? We suggest that modeling observation error as we have done here can address these questions and aid in the design of samples of species interactions. To try to survey to avoid all false-negatives is a fool’s errand. Species ranges overlap to form mosaics, which themselves are often changing in time. Communities and networks don’t end in space, and the interactions that connect species on the ‘periphery’ of a given network to species outside the spatial extent of a given sample will inevitably appear as false-negatives in practical samples. The goal should instead be to sample a system enough to have a statistically robust estimate of the current state and empirical change over time of an ecological community at a given spatial extent and temporal resolution, and to determine what the sampling effort required should be prior to sampling.</p>
            <h1 class="unnumbered" id="references">References</h1>
            <div id="refs" class="references csl-bib-body hanging-indent" role="doc-bibliography">
            <div id="ref-Banville2021ManJl" class="csl-entry" role="doc-biblioentry">
            Banville, F., Vissault, S. &amp; Poisot, T. (2021). Mangal.jl and EcologicalNetworks.jl: Two complementary packages for analyzing ecological networks in Julia. <em>Journal of Open Source Software</em>, 6, 2721.
            </div>
            <div id="ref-Bascompte2007PlaMut" class="csl-entry" role="doc-biblioentry">
            Bascompte, J. &amp; Jordano, P. (2007). Plant-Animal Mutualistic Networks: The Architecture of Biodiversity. <em>Annual Review of Ecology, Evolution, and Systematics</em>, 38, 567–593.
            </div>
            <div id="ref-Becker2021OptPre" class="csl-entry" role="doc-biblioentry">
            Becker, D.J., Albery, G.F., Sjodin, A.R., Poisot, T., Bergner, L.M., Dallas, T.A., <em>et al.</em> (2021). Optimizing predictive models to prioritize viral discovery in zoonotic reservoirs.
            </div>
            <div id="ref-Bezanson2015JulFre" class="csl-entry" role="doc-biblioentry">
            Bezanson, J., Edelman, A., Karpinski, S. &amp; Shah, V.B. (2015). Julia: A Fresh Approach to Numerical Computing.
            </div>
            <div id="ref-Blanchet2020CooNot" class="csl-entry" role="doc-biblioentry">
            Blanchet, F.G., Cazelles, K. &amp; Gravel, D. (2020). Co-occurrence is not evidence of ecological interactions. <em>Ecology Letters</em>, 23, 1050–1063.
            </div>
            <div id="ref-Blaom2020MljJul" class="csl-entry" role="doc-biblioentry">
            Blaom, A.D., Kiraly, F., Lienart, T., Simillides, Y., Arenas, D. &amp; Vollmer, S.J. (2020). MLJ: A Julia package for composable machine learning. <em>Journal of Open Source Software</em>, 5, 2704.
            </div>
            <div id="ref-Boakes2015InfSpe" class="csl-entry" role="doc-biblioentry">
            Boakes, E.H., Rout, T.M. &amp; Collen, B. (2015). Inferring species extinction: The use of sighting records. <em>Methods in Ecology and Evolution</em>, 6, 678–687.
            </div>
            <div id="ref-Canard2012EmeStr" class="csl-entry" role="doc-biblioentry">
            Canard, E., Mouquet, N., Marescot, L., Gaston, K.J., Gravel, D. &amp; Mouillot, D. (2012). Emergence of Structural Patterns in Neutral Trophic Networks. <em>PLOS ONE</em>, 7, e38295.
            </div>
            <div id="ref-CaraDonna2017IntRew" class="csl-entry" role="doc-biblioentry">
            CaraDonna, P.J., Petry, W.K., Brennan, R.M., Cunningham, J.L., Bronstein, J.L., Waser, N.M., <em>et al.</em> (2017). Interaction rewiring and the rapid turnover of plantpollinator networks. <em>Ecology Letters</em>, 20, 385–394.
            </div>
            <div id="ref-Carlson2020WhaWou" class="csl-entry" role="doc-biblioentry">
            Carlson, C.J., Dallas, T.A., Alexander, L.W., Phelan, A.L. &amp; Phillips, A.J. (2020). What would it take to describe the global diversity of parasites? <em>Proceedings of the Royal Society B: Biological Sciences</em>, 287, 20201841.
            </div>
            <div id="ref-Cazelles2016TheSpe" class="csl-entry" role="doc-biblioentry">
            Cazelles, K., Araújo, M.B., Mouquet, N. &amp; Gravel, D. (2016). A theory for species co-occurrence in interaction networks. <em>Theoretical Ecology</em>, 9, 39–48.
            </div>
            <div id="ref-Closs1994SpaTem" class="csl-entry" role="doc-biblioentry">
            Closs, G.P. &amp; Lake, P.S. (1994). Spatial and Temporal Variation in the Structure of an Intermittent-Stream Food Web. <em>Ecological Monographs</em>, 64, 1–21.
            </div>
            <div id="ref-deAguiar2019RevBia" class="csl-entry" role="doc-biblioentry">
            de Aguiar, M.A.M., Newman, E.A., Pires, M.M., Yeakel, J.D., Boettiger, C., Burkle, L.A., <em>et al.</em> (2019). Revealing biases in the sampling of ecological interaction networks. <em>PeerJ</em>, 7, e7566.
            </div>
            <div id="ref-Gauzens2013FooAgg" class="csl-entry" role="doc-biblioentry">
            Gauzens, B., Legendre, S., Lazzaro, X. &amp; Lacroix, G. (2013). Food-web aggregation, methodological and functional issues. <em>Oikos</em>, 122, 1606–1615.
            </div>
            <div id="ref-Giacomuzzo2021FooWeb" class="csl-entry" role="doc-biblioentry">
            Giacomuzzo, E. &amp; Jordán, F. (2021). Food web aggregation: Effects on key positions. <em>Oikos</em>, 130, 2170–2181.
            </div>
            <div id="ref-Griffiths1998SamEff" class="csl-entry" role="doc-biblioentry">
            Griffiths, D. (1998). Sampling effort, regression method, and the shape and slope of sizeabundance relations. <em>Journal of Animal Ecology</em>, 67, 795–804.
            </div>
            <div id="ref-Gupta2023HowMan" class="csl-entry" role="doc-biblioentry">
            Gupta, A., Figueroa H., D., O’Gorman, E., Jones, I., Woodward, G. &amp; Petchey, O.L. (2023). How many predator guts are required to predict trophic interactions? <em>Food Webs</em>, 34, e00269.
            </div>
            <div id="ref-Hadfield2014TalTwo" class="csl-entry" role="doc-biblioentry">
            Hadfield, J.D., Krasnov, B.R., Poulin, R. &amp; Nakagawa, S. (2014). A Tale of Two Phylogenies: Comparative Analyses of Ecological Interactions. <em>The American Naturalist</em>, 183, 174–187.
            </div>
            <div id="ref-Havens1992ScaStr" class="csl-entry" role="doc-biblioentry">
            Havens, K. (1992). Scale and Structure in Natural Food Webs. <em>Science</em>, 257, 1107–1109.
            </div>
            <div id="ref-Johnson2021BayEst" class="csl-entry" role="doc-biblioentry">
            Johnson, E.K. &amp; Larremore, D.B. (2021). Bayesian estimation of population size and overlap from random subsamples.
            </div>
            <div id="ref-Jordano2016SamNet" class="csl-entry" role="doc-biblioentry">
            Jordano, P. (2016). Sampling networks of ecological interactions. <em>Functional Ecology</em>, 30, 1883–1893.
            </div>
            <div id="ref-Joseph2020NeuHie" class="csl-entry" role="doc-biblioentry">
            Joseph, M.B. (2020). Neural hierarchical models of ecological populations. <em>Ecology Letters</em>, 23, 734–747.
            </div>
            <div id="ref-Kenall2014OpeFut" class="csl-entry" role="doc-biblioentry">
            Kenall, A., Harold, S. &amp; Foote, C. (2014). An open future for ecological and evolutionary data? <em>BMC Evolutionary Biology</em>, 14, 66.
            </div>
            <div id="ref-Khoshraftar2022SurGra" class="csl-entry" role="doc-biblioentry">
            Khoshraftar, S. &amp; An, A. (2022). A Survey on Graph Representation Learning Methods.
            </div>
            <div id="ref-Koleff2003MeaBet" class="csl-entry" role="doc-biblioentry">
            Koleff, P., Gaston, K.J. &amp; Lennon, J.J. (2003). Measuring beta diversity for presenceabsence data. <em>Journal of Animal Ecology</em>, 72, 367–382.
            </div>
            <div id="ref-Kopelke2017FooStr" class="csl-entry" role="doc-biblioentry">
            Kopelke, J.-P., Nyman, T., Cazelles, K., Gravel, D., Vissault, S. &amp; Roslin, T. (2017). Food-web structure of willow-galling sawflies and their natural enemies across Europe. <em>Ecology</em>, 98, 1730–1730.
            </div>
            <div id="ref-Li2021PerCom" class="csl-entry" role="doc-biblioentry">
            Li, M., Liu, R.-R., Lü, L., Hu, M.-B., Xu, S. &amp; Zhang, Y.-C. (2021). Percolation on complex networks: Theory and application. <em>Physics Reports</em>, Percolation on complex networks: Theory and application, 907, 1–68.
            </div>
            <div id="ref-MacDonald2020RevLin" class="csl-entry" role="doc-biblioentry">
            MacDonald, A.A.M., Banville, F. &amp; Poisot, T. (2020). Revisiting the Links-Species Scaling Relationship in Food Webs. <em>Patterns</em>, 1.
            </div>
            <div id="ref-Makiola2020KeyQue" class="csl-entry" role="doc-biblioentry">
            Makiola, A., Compson, Z.G., Baird, D.J., Barnes, M.A., Boerlijst, S.P., Bouchez, A., <em>et al.</em> (2020). Key Questions for Next-Generation Biomonitoring. <em>Frontiers in Environmental Science</em>, 7.
            </div>
            <div id="ref-Martinez1999EffSam" class="csl-entry" role="doc-biblioentry">
            Martinez, N.D., Hawkins, B.A., Dawah, H.A. &amp; Feifarek, B.P. (1999). Effects of Sampling Effort on Characterization of Food-Web Structure. <em>Ecology</em>, 80, 1044–1055.
            </div>
            <div id="ref-McLeod2021SamAsy" class="csl-entry" role="doc-biblioentry">
            McLeod, A., Leroux, S.J., Gravel, D., Chu, C., Cirtwill, A.R., Fortin, M.-J., <em>et al.</em> (2021). Sampling and asymptotic network properties of spatial multi-trophic networks. <em>Oikos</em>, 130, 2250–2259.
            </div>
            <div id="ref-Moore2016OptEco" class="csl-entry" role="doc-biblioentry">
            Moore, A.L. &amp; McCarthy, M.A. (2016). Optimizing ecological survey effort over space and time. <em>Methods in Ecology and Evolution</em>, 7, 891–899.
            </div>
            <div id="ref-Niedballa2019AssAna" class="csl-entry" role="doc-biblioentry">
            Niedballa, J., Wilting, A., Sollmann, R., Hofer, H. &amp; Courtiol, A. (2019). Assessing analytical methods for detecting spatiotemporal interactions between species from camera trapping data. <em>Remote Sensing in Ecology and Conservation</em>, 5, 272–285.
            </div>
            <div id="ref-Paine1988RoaMap" class="csl-entry" role="doc-biblioentry">
            Paine, R.T. (1988). Road Maps of Interactions or Grist for Theoretical Development? <em>Ecology</em>, 69, 1648–1654.
            </div>
            <div id="ref-Poisot2022GuiPre" class="csl-entry" role="doc-biblioentry">
            Poisot, T. (2022). Guidelines for the prediction of species interactions through binary classification.
            </div>
            <div id="ref-Poisot2021GloKno" class="csl-entry" role="doc-biblioentry">
            Poisot, T., Bergeron, G., Cazelles, K., Dallas, T., Gravel, D., MacDonald, A., <em>et al.</em> (2021). Global knowledge gaps in species interaction networks data. <em>Journal of Biogeography</em>, 48, 1552–1563.
            </div>
            <div id="ref-Poisot2022NetEmb" class="csl-entry" role="doc-biblioentry">
            Poisot, T., Ouellet, M.-A., Mollentze, N., Farrell, M.J., Becker, D.J., Brierly, L., <em>et al.</em> (2022). Network embedding unveils the hidden interactions in the mammalian virome.
            </div>
            <div id="ref-Poisot2015SpeWhy" class="csl-entry" role="doc-biblioentry">
            Poisot, T., Stouffer, D.B. &amp; Gravel, D. (2015). Beyond species: Why ecological interaction networks vary through space and time. <em>Oikos</em>, 124, 243–251.
            </div>
            <div id="ref-Ponisio2017OppAtt" class="csl-entry" role="doc-biblioentry">
            Ponisio, L.C., Gaiarsa, M.P. &amp; Kremen, C. (2017). Opportunistic attachment assembles plantpollinator networks. <em>Ecology Letters</em>, 20, 1261–1272.
            </div>
            <div id="ref-Savage2004EffBod" class="csl-entry" role="doc-biblioentry">
            Savage, V.M., Gillooly, J.F., Brown, J.H., West, G.B. &amp; Charnov, E.L. (2004). Effects of Body Size and Temperature on Population Growth. <em>The American Naturalist</em>, 163, 429–441.
            </div>
            <div id="ref-Song2022GenDri" class="csl-entry" role="doc-biblioentry">
            Song, C., Simmons, B.I., Fortin, M.-J. &amp; Gonzalez, A. (2022a). Generalism drives abundance: A computational causal discovery approach. <em>PLOS Computational Biology</em>, 18, e1010302.
            </div>
            <div id="ref-Song2022RapMon" class="csl-entry" role="doc-biblioentry">
            Song, C., Simmons, B.I., Fortin, M.-J., Gonzalez, A., Kaiser-Bunbury, C.N. &amp; Saavedra, S. (2022b). Rapid monitoring for ecological persistence.
            </div>
            <div id="ref-Stephenson2020TecAdv" class="csl-entry" role="doc-biblioentry">
            Stephenson, P. (2020). Technological advances in biodiversity monitoring: Applicability, opportunities and challenges. <em>Current Opinion in Environmental Sustainability</em>, Open issue 2020 part A: Technology Innovations and Environmental Sustainability in the Anthropocene, 45, 36–41.
            </div>
            <div id="ref-Stock2017LinFil" class="csl-entry" role="doc-biblioentry">
            Stock, M., Poisot, T., Waegeman, W. &amp; De Baets, B. (2017). Linear filtering reveals false negatives in species interaction data. <em>Scientific Reports</em>, 7, 45908.
            </div>
            <div id="ref-Strydom2021RoaPre" class="csl-entry" role="doc-biblioentry">
            Strydom, T., Catchen, M.D., Banville, F., Caron, D., Dansereau, G., Desjardins-Proulx, P., <em>et al.</em> (2021). A roadmap towards predicting species interaction networks (across space and time). <em>Philosophical Transactions of the Royal Society B: Biological Sciences</em>, 376, 20210063.
            </div>
            <div id="ref-Thompson2017DisGov" class="csl-entry" role="doc-biblioentry">
            Thompson, P.L. &amp; Gonzalez, A. (2017). Dispersal governs the reorganization of ecological networks under environmental change. <em>Nature Ecology &amp; Evolution</em>, 1, 1–8.
            </div>
            <div id="ref-Thompson2000ResSol" class="csl-entry" role="doc-biblioentry">
            Thompson, R.M. &amp; Townsend, C.R. (2000). Is resolution the solution?: The effect of taxonomic resolution on the calculated properties of three stream food webs. <em>Freshwater Biology</em>, 44, 413–422.
            </div>
            <div id="ref-Volkov2003NeuThe" class="csl-entry" role="doc-biblioentry">
            Volkov, I., Banavar, J.R., Hubbell, S.P. &amp; Maritan, A. (2003). Neutral theory and relative species abundance in ecology. <em>Nature</em>, 424, 1035–1037.
            </div>
            <div id="ref-Walther1995SamEff" class="csl-entry" role="doc-biblioentry">
            Walther, B.A., Cotgreave, P., Price, R.D., Gregory, R.D. &amp; Clayton, D.H. (1995). Sampling Effort and Parasite Species Richness. <em>Parasitology Today</em>, 11, 306–310.
            </div>
            <div id="ref-Williams2000SimRul" class="csl-entry" role="doc-biblioentry">
            Williams, R.J. &amp; Martinez, N.D. (2000). Simple rules yield complex food webs. <em>Nature</em>, 404, 180–183.
            </div>
            <div id="ref-Willott2001SpeAcc" class="csl-entry" role="doc-biblioentry">
            Willott, S.j. (2001). Species accumulation curves and the measure of sampling effort. <em>Journal of Applied Ecology</em>, 38, 484–486.
            </div>
            <div id="ref-Young2021RecPla" class="csl-entry" role="doc-biblioentry">
            Young, J.-G., Valdovinos, F.S. &amp; Newman, M.E.J. (2021). Reconstruction of plantpollinator networks from observational data. <em>Nature Communications</em>, 12, 3911.
            </div>
            <div id="ref-Zhang2019GraCon" class="csl-entry" role="doc-biblioentry">
            Zhang, S., Tong, H., Xu, J. &amp; Maciejewski, R. (2019). Graph convolutional networks: A comprehensive review. <em>Computational Social Networks</em>, 6, 11.
            </div>
            </div>
        </article>
    </main>

    <footer>
        <div class="container">
            <h2>Affiliations</h2> 
            <sup>1</sup>&nbsp;McGill University; <sup>2</sup>&nbsp;Québec Centre for Biodiversity Science; <sup>3</sup>&nbsp;Université de Montréal
            <br />
            
        </div>
    </footer>
    
</body>
</html>